### ISSUE TITLE
H100 - Failed setup, will try after update to 3.0.0

### ISSUE THREAD

**Issue by grzegorzsoltys** [petretiandrea/home-assistant-tapo-p100#717]
Hi,
after update to 3.0.0 I have a problem with hub H100. In settings and devices: Failed setup, will try:
![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/12548684/23175346-8fb1-47a8-a457-0fdd0145ff13)

H100 firmware version: Latest 1.5.6 Build 240202 Ref. 164142

HA:
Core 2024.2.5
Supervisor 2024.02.1
Operating System 12.0
Frontend 20240306.0

Logs:
This error originated from a custom integration.

Logger: plugp100.protocol.passthrough_protocol
Source: custom_components/tapo/coordinators.py:155
integration: TP-Link Tapo (documentation, issues)
First occurred: 06:51:51 (384 occurrences)
Last logged: 08:27:21

Session timeout, invalidate it, retrying with new session
![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/12548684/def24e4c-0bb9-4a80-9f09-616cfa558ca8)

**Comment by zee36** [comment]
Hi Everyone, same issue here - coming with the new 3.0.0 update.

**Comment by petretiandrea** [comment]
Please post logs here or try to re-add your device

**Comment by grzegorzsoltys** [comment]
Same error after deleting and adding. And log is only:
Source: custom_components/tapo/coordinators.py:155
integration: TP-Link Tapo (documentation, issues)

Session timeout, invalidate it, retrying with new session

**Comment by rfesig** [comment]
[home-assistant_tapo_2024-03-07T23-49-05.414Z.log](https://github.com/petretiandrea/home-assistant-tapo-p100/files/14531377/home-assistant_tapo_2024-03-07T23-49-05.414Z.log)

Here's a debug log file after deleting and recreating the device.

**Comment by petretiandrea** [comment]
> [home-assistant_tapo_2024-03-07T23-49-05.414Z.log](https://github.com/petretiandrea/home-assistant-tapo-p100/files/14531377/home-assistant_tapo_2024-03-07T23-49-05.414Z.log)
> 
> Here's a debug log file after deleting and recreating the device. I don't know if it's relevant or not, but I have two T300 water leak sensors that are not supported connected to the H100.

This log shows a working hub, right?

**Comment by rfesig** [comment]
> > [home-assistant_tapo_2024-03-07T23-49-05.414Z.log](https://github.com/petretiandrea/home-assistant-tapo-p100/files/14531377/home-assistant_tapo_2024-03-07T23-49-05.414Z.log)
> > Here's a debug log file after deleting and recreating the device. I don't know if it's relevant or not, but I have two T300 water leak sensors that are not supported connected to the H100.
> 
> This log shows a working hub, right?

Unfortunately, no. It cycles between connecting and failed to connect. The error message in the GUI shows the "Unexpected exception: 'NoneType' object has no attribute.

If it helps, I can expose a home assistant installation to the Internet and give you a login.

**Comment by PetriHakala** [comment]
I have the same problem after updating. Any update on a possible fix or a work around?

**Comment by grzegorzsoltys** [comment]
@PetriHakala for now rewert to 2.14.0

**Comment by rfesig** [comment]
I had some time to mess with this today. After removing the 2 T300 water leak sensors from the Tapo app, the integration loaded without error.  10 minutes later I re-added one of the T300s to the smart hub and the integration started failing again with the same error.

For now I'm going to use the 3.0 version and just leave the water leak sensors disconnected, as they aren't supported yet anyway.

**Comment by grzegorzsoltys** [comment]
@petretiandrea will this be fixed?

**Comment by rfesig** [comment]
I was only using this integration for an H100 hub and a few sensors. Previous to the 3.0.0 version it worked fine. After upgrading to the 3.0.0 version, I had to remove the T300 water leak sensors, or the integration would fail to load.

I converted over to using Matter for communication with the H100 hub earlier today, which allowed me to connect the T300 sensors back to it. The T300 sensors are not supported in Home Assistant, but at least I was able to connect them to the hub and now I can get alerts on my phone if a leak is detected.

The Matter integration doesn't show the battery status of the sensors, but I'm willing to accept that tradeoff. TP-Link is actively working on Matter integration, so maybe eventually all the functionality will be available to Home Assistant.
